opm/autodiff/GeoProps.hpp

```diff
@@ -197,6 +208,36 @@ namespace Opm
 
     };
 
+    template <class GridType>
+    inline void DerivedGeology::minPvFillProps_(const GridType &grid,
+                                                Opm::EclipseStateConstPtr eclState,
+                                                std::vector<double> &ntg)
+    {
+
+        int numCells = Opm::AutoDiffGrid::numCells(grid);
+        const int* global_cell = Opm::UgGridHelpers::globalCell(grid);
+        EclipseGridConstPtr eclgrid = eclState->getEclipseGrid();
+        std::vector<double> porv = eclState->getDoubleGridProperty("PORV")->getData();
+        for (int cellIdx = 0; cellIdx < numCells; ++cellIdx) {
+            const int nx = grid.cartdims[0];
+            const int ny = grid.cartdims[1];
+            const int cartesianCellIdx = global_cell[cellIdx];
+            const int cartesianCellIdxAbove = cartesianCellIdx - nx*ny;
+
+            // Average properties when the cell above exist and has porv less than the MINPV threshold
+            if ( cartesianCellIdxAbove > 0 && porv[cartesianCellIdxAbove] > 0 && porv[cartesianCellIdxAbove] < eclgrid->getMinpvValue() ) {
+
+                // Volume weighted arithmetic average of NTG
+                const double cellVolume = eclgrid->getCellVolume(cartesianCellIdx);
+                const double cellAboveVolume = eclgrid->getCellVolume(cartesianCellIdxAbove);
+                double inv_totalCellVolume = 1/(cellVolume + cellAboveVolume);
+
+                ntg[cartesianCellIdx] *= vol;
```

Comment:
I may be blind, but I cannot find a definition of `vol`. Should it be `cellVolume`?